pkg/kubelet/eviction/eviction_manager_test.go

```diff
@@ -67,39 +67,84 @@ func (m *mockImageGC) DeleteUnusedImages() (int64, error) {
+
+func makePodWithDiskStats(name string, requests api.ResourceList, limits api.ResourceList, rootFsUsed, logsUsed, perLocalVolumeUsed string) (*api.Pod, statsapi.PodStats) {
+	pod := newPod(name, []api.Container{
+		newContainer(name, requests, limits),
+	}, nil)
+	podStats := newPodDiskStats(pod, parseQuantity(rootFsUsed), parseQuantity(logsUsed), parseQuantity(perLocalVolumeUsed))
+	return pod, podStats
+}
+
+func makeMemoryStats(nodeAvailableBytes string, podStats map[*api.Pod]statsapi.PodStats) *statsapi.Summary {
+	val := resource.MustParse(nodeAvailableBytes)
+	availableBytes := uint64(val.Value())
+	WorkingSetBytes := uint64(val.Value())
+	result := &statsapi.Summary{
+		Node: statsapi.NodeStats{
+			Memory: &statsapi.MemoryStats{
+				AvailableBytes:  &availableBytes,
+				WorkingSetBytes: &WorkingSetBytes,
+			},
+		},
+		Pods: []statsapi.PodStats{},
 	}
-	summaryStatsMaker := func(nodeAvailableBytes string, podStats map[*api.Pod]statsapi.PodStats) *statsapi.Summary {
-		val := resource.MustParse(nodeAvailableBytes)
-		availableBytes := uint64(val.Value())
-		WorkingSetBytes := uint64(val.Value())
-		result := &statsapi.Summary{
-			Node: statsapi.NodeStats{
-				Memory: &statsapi.MemoryStats{
-					AvailableBytes:  &availableBytes,
-					WorkingSetBytes: &WorkingSetBytes,
+	for _, podStat := range podStats {
+		result.Pods = append(result.Pods, podStat)
+	}
+	return result
+}
+
+func makeDiskStats(rootFsAvailableBytes, imageFsAvailableBytes string, podStats map[*api.Pod]statsapi.PodStats) *statsapi.Summary {
+	rootFsVal := resource.MustParse(rootFsAvailableBytes)
+	rootFsBytes := uint64(rootFsVal.Value())
+	rootFsCapacityBytes := uint64(rootFsVal.Value() * 2)
+	imageFsVal := resource.MustParse(imageFsAvailableBytes)
+	imageFsBytes := uint64(imageFsVal.Value())
+	imageFsCapacityBytes := uint64(imageFsVal.Value() * 2)
+	result := &statsapi.Summary{
+		Node: statsapi.NodeStats{
+			Fs: &statsapi.FsStats{
+				AvailableBytes: &rootFsBytes,
+				CapacityBytes:  &rootFsCapacityBytes,
+			},
+			Runtime: &statsapi.RuntimeStats{
+				ImageFs: &statsapi.FsStats{
+					AvailableBytes: &imageFsBytes,
+					CapacityBytes:  &imageFsCapacityBytes,
 				},
 			},
-			Pods: []statsapi.PodStats{},
-		}
-		for _, podStat := range podStats {
-			result.Pods = append(result.Pods, podStat)
-		}
-		return result
+		},
+		Pods: []statsapi.PodStats{},
+	}
+	for _, podStat := range podStats {
+		result.Pods = append(result.Pods, podStat)
 	}
-	podsToMake := []struct {
-		name             string
-		requests         api.ResourceList
-		limits           api.ResourceList
-		memoryWorkingSet string
-	}{
+	return result
+}
+
+type podToMake struct {
+	name               string
+	requests           api.ResourceList
+	limits             api.ResourceList
+	memoryWorkingSet   string
+	rootFsUsed         string
+	logsFsUsed         string
+	perLocalVolumeUsed string
+}
+
+// TestMemoryPressure
+func TestMemoryPressure(t *testing.T) {
+	podMaker := makePodWithMemoryStats
```

Comment:
It would make sense, but most of the tests in this file start with defining podMaker and summaryStatsMaker, I didn't want to break this convention.